src/Symfony/Component/Messenger/Transport/Enhancers/MemoryLimitReceiver.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Enhancers;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Transport\ReceiverInterface;
+
+/**
+ * @author Simon Delicata <simon.delicata@free.fr>
+ */
+class MemoryLimitReceiver implements ReceiverInterface
+{
+    private $decoratedReceiver;
+    private $memoryLimit;
+    private $logger;
+    private $memoryResolver;
+
+    public function __construct(ReceiverInterface $decoratedReceiver, string $memoryLimit, LoggerInterface $logger = null, callable $memoryResolver = null)
+    {
+        $this->decoratedReceiver = $decoratedReceiver;
+        $this->memoryLimit = $this->convertToOctets($memoryLimit);
+        $this->logger = $logger;
+        $this->memoryResolver = $memoryResolver ?: function () {
+            return \memory_get_usage();
+        };
+    }
+
+    public function receive(callable $handler): void
+    {
+        $this->decoratedReceiver->receive(function ($message) use ($handler) {
+            $handler($message);
+
+            $memoryResolver = $this->memoryResolver;
+            if ($memoryResolver() >= $this->memoryLimit) {
+                $this->stop();
+                if (null !== $this->logger) {
+                    $this->logger->info('Receiver stopped due to memory limit of {limit} exceeded', array('limit' => $this->memoryLimit));
```

Comment:
Problably this willl be useful for debugging purpose, so `->debug()` fits better.

Reply:
I believe this is more than debug as it highlights an issue (i.e. too much memory consumed). So `info` is, to me, the ideal level.